src/core/options-watcher.cpp

```diff
@@ -0,0 +1,148 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2023 Intel Corporation. All Rights Reserved.
+
+
+#include <src/core/options-watcher.h>
+#include <proc/synthetic-stream.h>
+
+
+namespace librealsense {
+
+
+options_watcher::options_watcher( std::chrono::milliseconds update_interval )
+    : _update_interval( update_interval )
+    , _updater()
+{
+}
+
+options_watcher::~options_watcher()
+{
+    try
+    {
+        std::lock_guard< std::mutex > lock( _mutex );
+        _options.clear();
+        stop();
+    }
+    catch( ... )
+    {
+        LOG_DEBUG( "Failures taking lock or stoping thread while destructing options_watcher" );
+    }
+}
+
+void options_watcher::register_option( rs2_option id, std::shared_ptr< option > option )
```

Comment:
I think we should safeguard this: we shouldn't be able to un/register options once the thread is started...?

Reply:
I don't see a reason to prevent un/registration at any point. Guarded by mutex